instrumentation/CONTRIBUTING.md

```diff
@@ -0,0 +1,483 @@
+
+We welcome contributions from everyone, and we want to make sure that you have a great experience contributing to this project, as well as a positive impact on the Ruby community.
+
+We have limited capacity to maintain instrumentation libraries, so we ask that you commit to maintaining the instrumentation library you contribute.
+
+In addition to the requirements to maintain at least community member status, contributing an instrumentation to this project requires the following:
+
+1. Responding to issues and pull requests
+2. Performing timely code reviews and responding to issues
+3. Addressing security vulnerabilities
+4. Keeping the instrumentation library up to date with the latest:
+    * OpenTelemetry Ruby API and SDK changes
+    * Ruby language changes
+    * Instrumented library changes
+
+If you do not have the capacity to maintain the instrumentation library, please consider contributing to the OpenTelemetry Ruby project in other ways, or consider creating a separate project for the instrumentation library.
+
+> :warning: Libraries that do not meet these requirements may be removed from the project at any time at the discretion of OpenTelemetry Ruby Contrib Maintainers.
+
+## Contributing a New Instrumentation Library
+
+Our long-term goal is to provide instrumentation for all popular Ruby libraries. Ideally we would like to have first-party instrumentation for all libraries maintained by the Gem authors to ensure compatability with upstream gems, however in many cases this is not possible.
+
+For this reason we welcome contributions of new instrumentation libraries that cannot be maintained by the original gem authors as first party instrumentations.
+
+The following steps are required to contribute a new instrumentation library:
+
+1. Generate an instrumentation gem skeleton
+2. Implement the instrumentation library including comprehensive automated tests
+3. Add the instrumentation library to the appropriate CI workflow
+4. Include documentation for your instrumentation
+    * Document all instrumentation-specific configuration options in the `README.md` and `yardoc` comments
+    * Document all semantic conventions used by the instrumentation in the `README.md`
+    * Provide executable examples in an `examples` directory
+5. Submit a pull request
+
+## Generate the Gem
+
+This repository contains a script to generate a new instrumentation library.
+
+The snippet below demonstrates how to generate a an instrumentation for the `werewolf` gem, starting from the repository root.
+
+```console
+
+$bash opentelemetry-ruby-contrib> bin/instrumentation_generator werewolf
+
+```
+
+The output of the generator shows that it creates a new directory in the `instrumentation` directory using the name of the gem:
+
+``` console
+
+🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨🚨
+
+ WARNING: Your gem will *NOT* be tested until you add it to the CI workflows in `.github/workflows/ci.yml`!!
```

Comment:
We don't have a `ci.yml` file. Should this be updated to `.github/workflows/ci-instrumentation.yml`? 

We also put gems in `.github/dependabot.yml`. Should we call that out too? 

If you think script updates are outside the scope of this PR, I'm happy to make a follow-up issue to tackle these and anything else that comes up during the review.

Reply:
This is the output of the script. It is probably best to fix the script and then update this manual/guide.

Reply:
Sounds good. Here's the follow-up issue: https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/978 

You're right about mentioning the CI files. I hadn't gotten that far when I wrote this, but I think it  covers what's needed well.